File: BlogServer/Blog.Service/Api/CustomService.cs
using Blog.Model.Entity;
using Blog.Model.Param;
using Blog.Model.Rsult;
using Blog.Service.Api.Interface;
using SqlSugar;
namespace Blog.Service.Api
{
    public class CustomService: ICustomService
    {
        public CustomService(ISqlSugarClient client)
        {
            Db = client;
        }

        public ISqlSugarClient Db { get; }

        public async Task Insert(CustomInsertParam param)
        {
            var rsult = await Db.Queryable<CustomEnity>().Where(it => it.Name == param.Name).FirstAsync();
            if (rsult != null) throw new Exception("已存在同名项");
            var element = new CustomEnity
            {
                Content = param.Content,
                Clicks = 0,
                CreateDate = DateTime.Now,
                Name = param.Name
            };
            await Db.Storageable(element).ExecuteCommandAsync();
        }


        public async Task Update(CustomUpdateParam param)
        {
            var rsult = await Db.Queryable<CustomEnity>()
                .Where(it => it.Name == param.Name && it.Id != param.Id)
                .FirstAsync();
            if (rsult != null) throw new Exception("已存在同名项");
            var custom = await Db.Queryable<CustomEnity>().Where(it => it.Id == param.Id).FirstAsync();
            custom.Content = param.Content;
            custom.Name = param.Name;
            await Db.Storageable(custom).ExecuteCommandAsync();
        }

        public async Task<PageResult<CustomFindRsult>> List(CustomFindParam param)
        {
            RefAsync<int> total = 0;
            var list = await Db.Queryable<CustomEnity>()
                .WhereIF(!string.IsNullOrEmpty(param.Name), it => it.Name!.Contains(param.Name!))
                .ToPageListAsync(param.Page, param.PageNum, total);
            return new PageResult<CustomFindRsult>
            {
                Total = total,
                List = list.Select(x => new CustomFindRsult(x, false)).ToList()
            };
        }

        public async Task<CustomFindRsult> FindOne(IDParam param)
        {
            var custom = await Db.Queryable<CustomEnity>().Where(it => it.Id == param.Id).FirstAsync();
            if (custom == null) throw new NullReferenceException("该项不存在");
            return new CustomFindRsult(custom);
        }

        public async Task Delete(IDParam param)
        {
            await Db.Deleteable<CustomEnity>().In(it => it.Id, param.Id).ExecuteCommandAsync();
        }
    }
}

File: blog-server/src/module/admin/custom/Custom.service.ts
import {ConflictException, Injectable, NotFoundException} from '@nestjs/common'
import { InjectRepository } from '@nestjs/typeorm'
import { Like, Repository } from 'typeorm'
import { Custom } from '../../../db/entities/Custom'
import {CustomFindParam, CustomInsertParam, CustomUpdateParam} from "./model/Custom.param";
import {PageRsult} from "../../../model/BaseRsult";
import {IdParam} from "../../../model/BaseParam";
import {formatTime} from "../../../common/commonly.fun";


@Injectable()
export class CustomService {
    constructor(
        @InjectRepository(Custom) private customRepository: Repository<Custom>,
    ) {}

    // 新增
    async insert(body: CustomInsertParam): Promise<void> {
        const custom = await this.customRepository.findOne({where: { name: body.name }})
        if (custom) throw new ConflictException("已存在相同名称")
        const element = new Custom()
        element.content = body.content
        element.name = body.name
        element.createDate = formatTime()
        element.clicks = 0
        await this.customRepository.save(element)
    }


    // 修改
    async update(body: CustomUpdateParam): Promise<void> {
        const custom = await this.customRepository.findOne({where: { name: body.name }})
        if (!custom) throw new NotFoundException("项目不存在")
        custom.content = body.content
        custom.name = body.name
        await this.customRepository.save(custom)
    }

    // 查询
    async list(body: CustomFindParam): Promise<PageRsult<Custom>> {
        const list = await this.customRepository.find({
            select:[ 'id', 'name', 'createDate', 'clicks' ],
            where: { name: Like(`%${ body.name || '' }%`) },
            skip: body.pageNum * ( body.page - 1),
            take: body.pageNum
        })
        const total = await this.customRepository.count({where: { name: Like(`%${ body.name || '' }%`) }})
        return {
            list,
            total
        }
    }



    // 单个查询
    async findOne(body: IdParam): Promise<Custom> {
        const result = await this.customRepository.findOne({
            where: { id: body.id }
        })
        if (!result) throw new NotFoundException("项目不存在")
        return result
    }


    // 删除
    async delete(body: IdParam) :Promise<void> {
        await this.customRepository.delete({ id: body.id })
    }

}